polygeist/tools/cgeist/Lib/ValueCategory.cc

```diff
@@ -290,3 +290,59 @@ ValueCategory ValueCategory::FPExt(OpBuilder &Builder,
   warnUnconstrainedCast<arith::ExtFOp>();
   return Cast<arith::ExtFOp>(Builder, PromotionType);
 }
+
+ValueCategory ValueCategory::SIToFP(OpBuilder &Builder,
+                                    Type PromotionType) const {
+  warnUnconstrainedCast<arith::SIToFPOp>();
```

Comment:
This function (and related functions) expects the target type `PromotionType` to be a floating-point type. I'd add an assert to ensure that is the case.